Not able to build on macOS

Hi,
I am trying to build this project on macOS. It builds 98% successfully, OpenCV and TFLite also builds successfully. I have modifed the make file to use clang++ with -std=c++11 -stdlib=libc++. I have also commented out the code for v4l2(videoio) from deepseg.cc and related code from Makefile and CMakeLists.txt as I dont want that. 

```
$ clang++ --version
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```




Following is the error that I am getting. Could you please help resolve the issue?





```
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/optional_debug_tools.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/simple_memory_arena.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/stderr_reporter.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/string_util.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/tflite_with_xnnpack_optional.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/util.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/profiling/platform_profiler.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/schema/schema_utils.cc.o
[ 97%] Building CXX object tensorflow-lite/CMakeFiles/tensorflow-lite.dir/tools/optimize/sparsity/format_converter.cc.o
[ 98%] Linking CXX static library libtensorflow-lite.a
[ 98%] Built target tensorflow-lite
[ 98%] Building CXX object CMakeFiles/backscrub.dir/lib/libbackscrub.cc.o
[ 98%] Building CXX object CMakeFiles/backscrub.dir/lib/transpose_conv_bias.cc.o
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.cc:19:
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.h:18:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:23:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/compatibility.h:81:14: warning: 
      alias declarations are a C++11 extension [-Wc++11-extensions]
using int8 = std::int8_t;
             ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/compatibility.h:82:15: warning: 
      alias declarations are a C++11 extension [-Wc++11-extensions]
using uint8 = std::uint8_t;
              ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:152:10: error: 
      unknown type name 'constexpr'
  static constexpr int kMaxSmallSize = 5;
         ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:154:50: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  RuntimeShape& operator=(RuntimeShape const&) = delete;
                                                 ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:179:27: error: 
      no template named 'initializer_list' in namespace 'std'
  RuntimeShape(const std::initializer_list<int> init_list) : size_(0) {
                     ~~~~~^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:283:36: error: 
      no template named 'initializer_list' in namespace 'std'
  inline void BuildFrom(const std::initializer_list<int> init_list) {
                              ~~~~~^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:163:7: error: 
      use of undeclared identifier 'dims_pointer_'
      dims_pointer_ = new int32_t[dimensions_count];
      ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:190:7: error: 
      use of undeclared identifier 'dims_pointer_'
      dims_pointer_ = new int32_t[size_];
      ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:207:16: error: 
      use of undeclared identifier 'dims_pointer_'
      delete[] dims_pointer_;
               ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:216:36: error: 
      use of undeclared identifier 'dims_pointer_'
    return size_ > kMaxSmallSize ? dims_pointer_[i] : dims_[i];
                                   ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:216:55: error: 
      use of undeclared identifier 'dims_'; did you mean 'Dims'?
    return size_ > kMaxSmallSize ? dims_pointer_[i] : dims_[i];
                                                      ^~~~~
                                                      Dims
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:213:18: note: 
      'Dims' declared here
  inline int32_t Dims(int i) const {
                 ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:216:55: error: 
      reference to non-static member function must be called
    return size_ > kMaxSmallSize ? dims_pointer_[i] : dims_[i];
                                                      ^~~~~
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:222:7: error: 
      use of undeclared identifier 'dims_pointer_'; did you mean
      'std::is_pointer'?
      dims_pointer_[i] = val;
      ^~~~~~~~~~~~~
      std::is_pointer
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:817:50: note: 
      'std::is_pointer' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_pointer
                                                 ^
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.cc:19:
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.h:18:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:222:7: error: 
      use of undeclared identifier 'dims_pointer_'; did you mean
      'std::is_pointer'?
      dims_pointer_[i] = val;
      ^~~~~~~~~~~~~
      std::is_pointer
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:817:50: note: 
      'std::is_pointer' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_pointer
                                                 ^
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.cc:19:
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.h:18:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:222:7: error: 
      C++ requires a type specifier for all declarations
      dims_pointer_[i] = val;
      ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:224:7: error: 
      use of undeclared identifier 'dims_'
      dims_[i] = val;
      ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:229:36: error: 
      use of undeclared identifier 'dims_pointer_'
    return size_ > kMaxSmallSize ? dims_pointer_ : dims_;
                                   ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:229:52: error: 
      use of undeclared identifier 'dims_'; did you mean 'Dims'?
    return size_ > kMaxSmallSize ? dims_pointer_ : dims_;
                                                   ^~~~~
                                                   Dims
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:213:18: note: 
      'Dims' declared here
  inline int32_t Dims(int i) const {
                 ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:232:36: error: 
      use of undeclared identifier 'dims_pointer_'
    return size_ > kMaxSmallSize ? dims_pointer_ : dims_;
                                   ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:232:52: error: 
      use of undeclared identifier 'dims_'; did you mean 'Dims'?
    return size_ > kMaxSmallSize ? dims_pointer_ : dims_;
                                                   ^~~~~
                                                   Dims
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:213:18: note: 
      'Dims' declared here
  inline int32_t Dims(int i) const {
                 ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:235:57: error: 
      use of undeclared identifier 'dims_'; did you mean 'Dims'?
  inline const int32_t* DimsDataUpTo5D() const { return dims_; }
                                                        ^~~~~
                                                        Dims
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:213:18: note: 
      'Dims' declared here
  inline int32_t Dims(int i) const {
                 ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:235:57: error: 
      reference to non-static member function must be called
  inline const int32_t* DimsDataUpTo5D() const { return dims_; }
                                                        ^~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
12 warnings and 20 errors generated.
make[2]: *** [CMakeFiles/backscrub.dir/lib/transpose_conv_bias.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/libbackscrub.cc:5:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/interpreter.h:34:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/allocation.h:35:8: warning: 
      scoped enumerations are a C++11 extension [-Wc++11-extensions]
  enum class Type {
       ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/allocation.h:72:28: warning: 
      'override' keyword is a C++11 extension [-Wc++11-extensions]
  const void* base() const override;
                           ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/allocation.h:73:24: warning: 
      'override' keyword is a C++11 extension [-Wc++11-extensions]
  size_t bytes() const override;
                       ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/allocation.h:74:22: warning: 
      'override' keyword is a C++11 extension [-Wc++11-extensions]
  bool valid() const override;
                     ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/allocation.h:82:16: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  int mmap_fd_ = -1;  // mmap file descriptor
               ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/experimental/resource/resource_base.h:38:66: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
    std::unordered_map<std::int32_t, std::unique_ptr<ResourceBase>>;
                                                                 ^~
                                                                 > >
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/libbackscrub.cc:5:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/interpreter.h:38:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:35:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:39:1: error: 
      unknown type name 'constexpr'
constexpr int kDefaultTensorAlignment = 64;
^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:45:1: error: 
      unknown type name 'constexpr'
constexpr char kFlexCustomCodePrefix[] = "Flex";
^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:65:27: error: 
      no member named 'initializer_list' in namespace 'std'
size_t CombineHashes(std::initializer_list<size_t> hashes);
                     ~~~~~^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:65:44: error: 
      unexpected type name 'size_t': expected expression
size_t CombineHashes(std::initializer_list<size_t> hashes);
                                           ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:65:52: error: 
      use of undeclared identifier 'hashes'
size_t CombineHashes(std::initializer_list<size_t> hashes);
                                                   ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/util.h:97:1: error: 
      unknown type name 'constexpr'
constexpr char kValidationSubgraphNamePrefix[] = "VALIDATION:";
^
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/libbackscrub.cc:5:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/interpreter.h:38:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:50:48: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
           std::vector<std::unique_ptr<Subgraph>>* subgraphs,
                                               ^~
                                               > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:197:61: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
  const std::vector<std::pair<TfLiteNode, TfLiteRegistration>>&
                                                            ^~
                                                            > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:291:10: error: 
      unknown type name 'constexpr'
  static constexpr int kTensorsReservedCapacity = 128;
         ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:296:10: error: 
      unknown type name 'constexpr'
  static constexpr int kTensorsCapacityHeadroom = 16;
         ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:320:39: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
  std::vector<std::unique_ptr<Subgraph>>* GetSubgraphs() { return sub...
                                      ^~
                                      > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:366:30: warning: 
      'override' keyword is a C++11 extension [-Wc++11-extensions]
    ~SubgraphAwareProfiler() override {}
                             ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:646:55: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
  std::vector<std::pair<TfLiteNode, TfLiteRegistration>>
                                                      ^~
                                                      > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:652:20: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool consistent_ = true;
                   ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:707:26: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool delegates_undone_ = false;
                         ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:719:52: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
  std::vector<std::pair<int, TfLiteCustomAllocation>> custom_allocations_;
                                                   ^~
                                                   > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:724:40: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool tensor_resized_since_op_invoke_ = false;
                                       ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:730:39: error: 
      a space is required between consecutive right angle brackets
      (use '> >')
  std::vector<std::unique_ptr<Subgraph>>* subgraphs_ = nullptr;
                                      ^~
                                      > >
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:730:54: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  std::vector<std::unique_ptr<Subgraph>>* subgraphs_ = nullptr;
                                                     ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:736:29: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool has_dynamic_tensors_ = true;
                            ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:741:40: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool (*check_cancelled_func_)(void*) = nullptr;
                                       ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:745:28: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  void* cancellation_data_ = nullptr;
                           ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:748:37: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  resource::ResourceMap* resources_ = nullptr;
                                    ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:755:30: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool preserve_all_tensors_ = false;
                             ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:759:28: warning: 
      in-class initialization of non-static data member is a C++11 extension
      [-Wc++11-extensions]
  bool is_subgraph_in_use_ = false;
                           ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/core/subgraph.h:124:74: error: 
      expected expression
  ...is_variable = false, const std::vector<int>& dims_signature = {}) {
                                                                   ^
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/libbackscrub.cc:5:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/interpreter.h:40:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/external_cpu_backend_context.h:99:65: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
  ExternalCpuBackendContext(const ExternalCpuBackendContext&) = delete;
                                                                ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/external_cpu_backend_context.h:101:7: warning: 
      deleted function definitions are a C++11 extension [-Wc++11-extensions]
      delete;
      ^
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/libbackscrub.cc:5:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/interpreter.h:42:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/portable_type_to_tflitetype.h:39:1: error: 
      unknown type name 'constexpr'
constexpr TfLiteType typeToTfLiteType() {
^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/portable_type_to_tflitetype.h:39:11: warning: 
      variable templates are a C++14 extension [-Wc++14-extensions]
constexpr TfLiteType typeToTfLiteType() {
          ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/portable_type_to_tflitetype.h:39:21: error: 
      expected ';' at end of declaration
constexpr TfLiteType typeToTfLiteType() {
                    ^
                    ;
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/portable_type_to_tflitetype.h:39:22: error: 
      C++ requires a type specifier for all declarations
constexpr TfLiteType typeToTfLiteType() {
                     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
44 warnings and 20 errors generated.
make[2]: *** [CMakeFiles/backscrub.dir/lib/libbackscrub.cc.o] Error 1
make[1]: *** [CMakeFiles/backscrub.dir/all] Error 2
make: *** [all] Error 2
Navid-Mac:build maggarwal$ 
```

Following dependencies were built successfully:

make -j 2
[  0%] Built target pthreadpool
[  0%] Built target absl_log_severity
[  0%] Built target absl_spinlock_wait
[  0%] Built target absl_dynamic_annotations
[  0%] Built target absl_time_zone
[  0%] Built target absl_int128
[  0%] Built target absl_civil_time
[  1%] Built target absl_city
[  1%] Built target farmhash
[  1%] Built target fft2d_fftsg
[  1%] Built target flatbuffers
[  2%] Built target ruy
[  2%] Built target clog
[  2%] Built target absl_raw_logging_internal
[  2%] Built target fft2d_fftsg2d
[  3%] Built target cpuinfo
[  3%] Built target absl_base
[  3%] Built target absl_throw_delegate
[  3%] Built target absl_debugging_internal
[  3%] Built target absl_bad_optional_access
[  3%] Built target absl_strings_internal
[  3%] Built target absl_bad_variant_access
[  3%] Built target absl_malloc_internal
[  3%] Built target absl_demangle_internal
[  3%] Built target absl_stacktrace
[  4%] Built target absl_strings
[  4%] Built target absl_graphcycles_internal
[  5%] Built target absl_time
[  5%] Built target absl_str_format_internal
[  5%] Built target absl_hash
[  5%] Built target absl_cord
[  5%] Built target absl_symbolize
[  5%] Built target absl_flags_marshalling
[  6%] Built target absl_status
[  6%] Built target absl_synchronization
[ 60%] Built target absl_flags_program_name
[ 88%] Built target XNNPACK
[ 89%] Built target absl_flags_config
[ 89%] Built target absl_flags_registry
[ 89%] Built target absl_flags_internal
[ 89%] Built target absl_flags
[ 98%] Built target tensorflow-lite

Getting error after this:

[ 98%] Building CXX object CMakeFiles/backscrub.dir/lib/libbackscrub.cc.o
[ 98%] Building CXX object CMakeFiles/backscrub.dir/lib/transpose_conv_bias.cc.o
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.cc:19:
In file included from /Users/maggarwal/Desktop/Work/backscrub/lib/transpose_conv_bias.h:18:
In file included from /Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/types.h:23:
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/compatibility.h:81:14: warning: alias declarations
      are a C++11 extension [-Wc++11-extensions]
using int8 = std::int8_t;
             ^
/Users/maggarwal/Desktop/Work/backscrub/tensorflow/tensorflow/lite/kernels/internal/compatibility.h:82:15: warning: alias declarations
      are a C++11 extension [-Wc++11-extensions]
using uint8 = std::uint8_t;
              ^

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 3.16)
 project(backscrub CXX)
 
+# globally set C++17 as default for compilation
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
 # allow override of Tensorflow location
 if(NOT DEFINED TENSORFLOW)
   set(TENSORFLOW tensorflow)

